src/taipy/core/submission/submission.py

```diff
@@ -44,19 +44,30 @@ class Submission(_Entity, _Labeled):
     def __init__(
         self,
         entity_id: str,
+        entity_type: str,
         id: Optional[str] = None,
         jobs: Optional[Union[List[Job], List[JobId]]] = None,
         creation_date: Optional[datetime] = None,
         submission_status: Optional[SubmissionStatus] = None,
         version: Optional[str] = None,
     ):
         self._entity_id = entity_id
+        self._entity_type = entity_type
         self.id = id or self.__new_id()
-        self._jobs: Union[List[Job], List[JobId], List] = jobs or []
+        self._job_ids = [job if isinstance(job, str) else job.id for job in jobs] if jobs else []
         self._creation_date = creation_date or datetime.now()
         self._submission_status = submission_status or SubmissionStatus.SUBMITTED
         self._version = version or _VersionManagerFactory._build_manager()._get_latest_version()
 
+        self.__abandoned = False
+        self.__canceled = False
+        self.__completed = False
+        self.__failed = False
+
+        self.__running_jobs: MutableSet[str] = set()
+        self.__blocked_jobs: MutableSet[str] = set()
+        self.__pending_jobs: MutableSet[str] = set()
+
```

Comment:
- It won't be persisted on the database so that we can avoid loading time.
- concurrent access is an interesting point, I think we should be safe but since there will be writing to the "job sets",  wouldnt hurt to put in some locking mechanism just in case. What do you think? @gmarabout

Reply:
I believe the discussion has been made and further talk is linked in the comment below